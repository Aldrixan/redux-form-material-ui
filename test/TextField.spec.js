import React from "react"
import ReduxFormMaterialUITextField from "../src/TextField"
import noop from "lodash.noop"
import renderer from "react-test-renderer"

describe("TextField", () => {
    it("renders a TextField", () => {
        const component = renderer.create(
            <ReduxFormMaterialUITextField input={{ onChange: noop, name: "myText", value: "Foo" }} meta={{}} />
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it("renders a TextField with no error when not touched", () => {
        const component = renderer.create(
            <ReduxFormMaterialUITextField
                input={{ onChange: noop, name: "myText", value: "Foo" }}
                meta={{ error: "FooError" }}
            />
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it("renders a TextField with an error", () => {
        const component = renderer.create(
            <ReduxFormMaterialUITextField
                input={{ onChange: noop, name: "myText", value: "Foo" }}
                meta={{ error: "FooError", touched: true }}
            />
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it("renders a TextField with no warning when not touched", () => {
        const component = renderer.create(
            <ReduxFormMaterialUITextField
                input={{ onChange: noop, name: "myText", value: "Foo" }}
                meta={{ error: "FooWarning" }}
            />
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it("renders a TextField with an warning", () => {
        const component = renderer.create(
            <ReduxFormMaterialUITextField
                input={{ onChange: noop, name: "myText", value: "Foo" }}
                meta={{ error: "FooWarning", touched: true }}
            />
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it("should ignore defaultValue", () => {
        const component = renderer.create(
            <ReduxFormMaterialUITextField
                input={{ onChange: noop, name: "myText", value: "Foo" }}
                defaultValue="5"
                meta={{ error: "FooWarning", touched: true }}
            />
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
