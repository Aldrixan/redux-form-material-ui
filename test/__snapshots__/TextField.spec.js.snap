// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField renders a TextField 1`] = `
<div
  className="MuiFormControl-root-1"
>
  <div
    className="MuiInput-root-5 MuiInput-formControl-6 MuiInput-underline-9"
  >
    <input
      aria-invalid={false}
      className="MuiInput-input-13"
      disabled={false}
      name="myText"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value="Foo"
    />
  </div>
</div>
`;

exports[`TextField renders a TextField with an error 1`] = `
<div
  className="MuiFormControl-root-1"
>
  <div
    className="MuiInput-root-5 MuiInput-formControl-6 MuiInput-underline-9"
  >
    <input
      aria-invalid={false}
      className="MuiInput-input-13"
      disabled={false}
      name="myText"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value="Foo"
    />
  </div>
  <p
    className="MuiFormHelperText-root-18 MuiFormHelperText-filled-23"
  >
    FooError
  </p>
</div>
`;

exports[`TextField renders a TextField with an warning 1`] = `
<div
  className="MuiFormControl-root-1"
>
  <div
    className="MuiInput-root-5 MuiInput-formControl-6 MuiInput-underline-9"
  >
    <input
      aria-invalid={false}
      className="MuiInput-input-13"
      disabled={false}
      name="myText"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value="Foo"
    />
  </div>
  <p
    className="MuiFormHelperText-root-18 MuiFormHelperText-filled-23"
  >
    FooWarning
  </p>
</div>
`;

exports[`TextField renders a TextField with no error when not touched 1`] = `
<div
  className="MuiFormControl-root-1"
>
  <div
    className="MuiInput-root-5 MuiInput-formControl-6 MuiInput-underline-9"
  >
    <input
      aria-invalid={false}
      className="MuiInput-input-13"
      disabled={false}
      name="myText"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value="Foo"
    />
  </div>
</div>
`;

exports[`TextField renders a TextField with no warning when not touched 1`] = `
<div
  className="MuiFormControl-root-1"
>
  <div
    className="MuiInput-root-5 MuiInput-formControl-6 MuiInput-underline-9"
  >
    <input
      aria-invalid={false}
      className="MuiInput-input-13"
      disabled={false}
      name="myText"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value="Foo"
    />
  </div>
</div>
`;

exports[`TextField should ignore defaultValue 1`] = `
<div
  className="MuiFormControl-root-1"
>
  <div
    className="MuiInput-root-5 MuiInput-formControl-6 MuiInput-underline-9"
  >
    <input
      aria-invalid={false}
      className="MuiInput-input-13"
      disabled={false}
      name="myText"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value="Foo"
    />
  </div>
  <p
    className="MuiFormHelperText-root-18 MuiFormHelperText-filled-23"
  >
    FooWarning
  </p>
</div>
`;
